app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
         assertTrue(hintMessage.waitForExists(waitingTime))
     }
 
+    fun clickForm(
+        formType: String,
+        calendarForm: Boolean = false,
+        clockForm: Boolean = false,
+        colorPicker: Boolean = false,
+    ) {
+        mDevice.waitForObjects(webPageItemContainingText(formType))
+
+        if (calendarForm) {
+            clickPageObject(webPageItemWithResourceId("calendar"))
+            mDevice.waitForIdle(waitingTime)
+        } else if (clockForm) {
+            clickPageObject(webPageItemWithResourceId("clock"))
+            mDevice.waitForIdle(waitingTime)
+        } else if (colorPicker) {
+            clickPageObject(webPageItemWithResourceId("colorPicker"))
+            mDevice.waitForIdle(waitingTime)
+        } else {
+            clickPageObject(webPageItemWithResourceId("dropDown"))
+            mDevice.waitForIdle(waitingTime)
+        }
+    }
+
+    fun clickFormViewButton(button: String) {
+        val clockAndCalendarButton = mDevice.findObject(UiSelector().textContains(button))
+        clockAndCalendarButton.click()
+    }
+
+    fun selectDate() {
+        mDevice.findObject(UiSelector().resourceId("android:id/month_view")).waitForExists(waitingTime)
+
+        val monthViewerCurrentDay =
+            mDevice.findObject(
+                UiSelector()
+                    .textContains("$currentDay")
+                    .descriptionContains("$currentDay $currentMonth $currentYear"),
+            )
+
+        monthViewerCurrentDay.click()
```

Comment:
same as above, maybe not be worth defining the var if it's not re-usable, it just adds more code.

Reply:
Done !